Sources_v3/Controllers/ChannelController.swift

```diff
@@ -99,42 +105,65 @@ public class ChannelControllerGeneric<ExtraData: ExtraDataTypes>: Controller, De
             stateDelegate = anyDelegate
         }
     }
-    
-    private(set) lazy var channelObserver: EntityDatabaseObserver<ChannelModel<ExtraData>, ChannelDTO> = {
-        let observer = EntityDatabaseObserver(context: self.client.databaseContainer.viewContext,
-                                              fetchRequest: ChannelDTO.fetchRequest(for: self.channelQuery.cid),
-                                              itemCreator: ChannelModel<ExtraData>.create)
-        observer.onChange = { [unowned self] change in
-            self.delegateCallback { $0?.channelController(self, didUpdateChannel: change) }
-        }
-        
-        return observer
-    }()
-    
-    private(set) lazy var messagesObserver: ListDatabaseObserver<MessageModel<ExtraData>, MessageDTO> = {
-        let observer = ListDatabaseObserver(context: self.client.databaseContainer.viewContext,
-                                            fetchRequest: MessageDTO.messagesFetchRequest(for: self.channelQuery.cid),
-                                            itemCreator: MessageModel<ExtraData>.init)
-        observer.onChange = { [unowned self] changes in
-            self.delegateCallback {
-                $0?.channelController(self, didUpdateMessages: changes)
-            }
-        }
-        
-        return observer
-    }()
+
+    @Cached private var channelObserver: EntityDatabaseObserver<ChannelModel<ExtraData>, ChannelDTO>
+    @Cached private var messagesObserver: ListDatabaseObserver<MessageModel<ExtraData>, MessageDTO>
     
     private let environment: Environment
+
+    private var isChannelCreated: Bool
+    private var channelCreatedCallback: ((ChannelId) -> Void)? {
+        isChannelCreated ? nil : { [weak self] cid in
+            guard let self = self else { return }
+            self.isChannelCreated = true
+            self.channelQuery = ChannelQuery(cid: cid, channelQuery: self.channelQuery)
+        }
+    }
     
     /// Creates a new `ChannelController`
     /// - Parameters:
     ///   - channelQuery: channel query for observing changes
     ///   - client: The `Client` this controller belongs to.
     ///   - environment: Envrionment for this controller.
-    init(channelQuery: ChannelQuery<ExtraData>, client: Client<ExtraData>, environment: Environment = .init()) {
+    ///   - isChannelCreated: Flag indicating whether channel is created on backend.
+    init(
+        channelQuery: ChannelQuery<ExtraData>,
+        client: Client<ExtraData>,
+        environment: Environment = .init(),
+        isChannelCreated: Bool = true
```

Comment:
I have a hard time wrapping my head around this name. Suggestions: `isNewChannel`, `isChannelAlreadyCreated`. I think I like the flipped `isNewChannel` version the most but I understand it doesn't fit the internal usage the best.

Reply:
You reacted 👍 but didn't change anything? :)

Reply:
Haha, sorry. Changed class property name but missed init signature naming :)